lib/boxcars/boxcar/vector_answer.rb

```diff
@@ -0,0 +1,58 @@
+# frozen_string_literal: true
+
+# Boxcars is a framework for running a series of tools to get an answer to a question.
+module Boxcars
+  # A Boxcar that interprets a prompt and executes ruby code to do math
+  class VectorAnswer < EngineBoxcar
+    # the description of this engine boxcar
+    DESC = "useful for when you need to answer questions from vector search results."
+
+    attr_reader :embeddings, :vector_documents, :search_content
+
+    # @param embeddings [Hash] The vector embeddings to use for this boxcar.
+    # @param vector_documents [Hash] The vector documents to use for this boxcar.
+    # @param engine [Boxcars::Engine] The engine to user for this boxcar. Can be inherited from a train if nil.
+    # @param prompt [Boxcars::Prompt] The prompt to use for this boxcar. Defaults to built-in prompt.
+    # @param kwargs [Hash] Any other keyword arguments to pass to the parent class.
+    def initialize(embeddings:, vector_documents:, engine: nil, prompt: nil, **kwargs)
+      the_prompt = prompt || my_prompt
+      @embeddings = embeddings
+      @vector_documents = vector_documents
```

Comment:
please check spec/boxcars/vector_search_spec.rb for this. because we have multiple ways for vector stores, the configuration may differ.

Reply:
I will take a look, and was wondering how to handle the multiple types.

Reply:
can we add a spec for this? we can use in_memory with simple array https://github.com/BoxcarsAI/boxcars/blob/main/spec/boxcars/vector_store/in_memory/search_spec.rb#L13

Reply:
@jaigouk - I didn't change anything yet. I will open another ticket to make this simpler.